Microsoft.Identity.Web/Client/TokenAcquisition.cs

```diff
@@ -62,12 +62,17 @@ public class TokenAcquisition : ITokenAcquisition
         /// <param name="configuration"></param>
         public TokenAcquisition(IConfiguration configuration, IMSALAppTokenCacheProvider appTokenCacheProvider, IMSALUserTokenCacheProvider userTokenCacheProvider)
         {
+            if (configuration == null)
+                throw new ArgumentNullException(nameof(configuration));
+
             azureAdOptions = new AzureADOptions();
             configuration.Bind("AzureAD", azureAdOptions);
 
             _applicationOptions = new ConfidentialClientApplicationOptions();
             configuration.Bind("AzureAD", _applicationOptions);
 
+            // brentsch - is it OK for these to be null?
+            // what programming model does this support?
```

Comment:
@brentschmaltz 
If the confidential client only uses flows about users (Authorization code flow, OBO), then the app token cache provider can be null. If the confidential client only uses client creds, the user token cache can be null.